validator/sawtooth_validator/gossip/signature_verifier.py

```diff
@@ -93,6 +94,16 @@ def validate_batch(batch):
                              batch.header_signature,
                              txn.header_signature)
                 valid = False
+                return valid
+
+            # verify the payload field matches the header
+            txn_payload_sha512 = hashlib.sha512(txn.payload).hexdigest()
+            if txn_payload_sha512 != txn_header.payload_sha512:
+                LOGGER.debug("the payload field doesn't match the header"
```

Comment:
I think "payload doesn't match payload_sha512 of the header for txn" is much more detailed.